mera-weather-app/src/hooks/use-http.js

```diff
@@ -0,0 +1,36 @@
+import { useState, useCallback } from "react";
+
+const useHttp = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(null);
+
+  const sendRequest = useCallback(async (requestConfig, applyData) => {
+    setIsLoading(true);
+    setError(null);
+    try {
+      const response = await fetch(requestConfig.url, {
+        method: requestConfig.method ? requestConfig.method : "GET",
+        headers: requestConfig.headers ? requestConfig.headers : {},
```

Comment:
A bit cleaner, in my opinion:
```
const {
   method = "GET",
   headers = {}
} = requestConfig;
```
and then you can directly use them in the request options object.